src/explorer/AppSettingsNodes.ts

```diff
@@ -35,39 +40,41 @@ export class AppSettingsNode extends NodeBase {
             this.subscription.subscriptionId);
     }
 
-    getTreeItem(): TreeItem {
+    public getTreeItem(): TreeItem {
         return {
             label: this.label,
             collapsibleState: TreeItemCollapsibleState.Collapsed,
-            contextValue: "applicationSettings",
+            contextValue: `applicationSettings`,
             iconPath: {
                 light: path.join(__filename, '..', '..', '..', '..', 'resources', 'light', 'Settings_16x_vscode.svg'),
                 dark: path.join(__filename, '..', '..', '..', '..', 'resources', 'dark', 'Settings_16x_vscode.svg')
             }
-        }
+        };
     }
 
-    async getChildren(): Promise<NodeBase[]> {
+    public async getChildren(): Promise<NodeBase[]> {
         const webApps = this.WebSiteManagementClient.webApps;
         const children: AppSettingNode[] = [];
         this._settings = this._isSlot ? await webApps.listApplicationSettingsSlot(this.site.resourceGroup, this._siteName, this._slotName) :
             await webApps.listApplicationSettings(this.site.resourceGroup, this._siteName);
 
         if (this._settings.properties) {
-            for (let key in this._settings.properties) {
-                children.push(new AppSettingNode(
-                    key,
-                    this._settings.properties[key],
-                    this.getTreeDataProvider(),
-                    this
-                ));
+            for (const key in this._settings.properties) {
+                if (this._settings.properties[key]) {
```

Comment:
This is a functionality change, right?  It's valid to have a property that's an empty string, so this seems wrong...?

(BTW, not a good idea to put functionality changes into a PR like this.)

Reply:
Yeah I think the correct fix would be to make 'value' an optional string in the AppSettingNode constructor

Reply:
A value should always be required for an app setting. Can it be:
```
const value = this._settings.properties[key] ? this._settings.properties[key] : '';
children.push(new AppSettingNode(
    key,
    value,
    this.getTreeDataProvider(),
    this
));
```

Reply:
It was the change recommended by TSLint for for-in array loops.  Ideally, they recommend not using for-in loops on arrays in general.  However, after speaking with Eric, I believe that we would want to use a for-of loop anyway.

https://stackoverflow.com/questions/29285897/what-is-the-difference-between-for-in-and-for-of-in-javascript

Should I make that change in my PR or hold off?

Reply:
I'm using Jay's implementation for now.  I had to disable the "forin" tslint rule which states: "Requires a `for ... in` statement to be filtered with an `if` statement."

Reply:
You are correct that it requires an if statement, but it requires a specific _type_ of if statement, which should always apply:
```
if (this._settings.properties.hasOwnProperty(key)) {
```

It's used to 'Prevents accidental iteration over properties inherited from an object’s prototype.'

Reply:
I would still prefer to keep 'no-for-in' and 'forin' rules. I believe this would satisfy both rules:
```
Object.keys(this._settings.properties).forEach((key: string) => {
    children.push(new AppSettingNode(
        key,
        this._settings.properties[key],
        this.getTreeDataProvider(),
        this
    ));
});
```

Reply:
+1